test/test_functional_tensor.py

```diff
@@ -692,6 +433,250 @@ def test_scale_channel(self):
         assert_equal(scaled_cpu, scaled_cuda.to('cpu'))
 
 
+class TestAffine:
+
+    @pytest.mark.parametrize('device', cpu_and_gpu())
+    @pytest.mark.parametrize('height, width', [(26, 26), (32, 26)])
+    @pytest.mark.parametrize('dt', [None, torch.float32, torch.float64, torch.float16])
+    def test_identity_map(self, device, height, width, dt):
+        # Tests on square and rectangular images
+        scripted_affine = torch.jit.script(F.affine)
+        tensor, pil_img = _create_data(height, width, device=device)
+
+        if dt == torch.float16 and device == "cpu":
+            # skip float16 on CPU case
+            return
+
+        if dt is not None:
+            tensor = tensor.to(dtype=dt)
+
+        # 1) identity map
+        out_tensor = F.affine(tensor, angle=0, translate=[0, 0], scale=1.0, shear=[0.0, 0.0], interpolation=NEAREST)
+
+        assert_equal(tensor, out_tensor, msg="{} vs {}".format(out_tensor[0, :5, :5], tensor[0, :5, :5]))
+        out_tensor = scripted_affine(
+            tensor, angle=0, translate=[0, 0], scale=1.0, shear=[0.0, 0.0], interpolation=NEAREST
+        )
+        assert_equal(tensor, out_tensor, msg="{} vs {}".format(out_tensor[0, :5, :5], tensor[0, :5, :5]))
+
+    @pytest.mark.parametrize('device', cpu_and_gpu())
+    @pytest.mark.parametrize('height, width', [(26, 26)])
+    @pytest.mark.parametrize('dt', [None, torch.float32, torch.float64, torch.float16])
+    @pytest.mark.parametrize('angle, config', [
+        (90, {'k': 1, 'dims': (-1, -2)}),
+        (45, None),
+        (30, None),
+        (-30, None),
+        (-45, None),
+        (-90, {'k': -1, 'dims': (-1, -2)}),
+        (180, {'k': 2, 'dims': (-1, -2)}),
+    ])
+    @pytest.mark.parametrize('scripted_affine', [True, False])
+    def test_square_rotations(self, device, height, width, dt, angle, config, scripted_affine):
+        # 2) Test rotation
+        tensor, pil_img = _create_data(height, width, device=device)
+
+        if dt == torch.float16 and device == "cpu":
+            # skip float16 on CPU case
+            return
+
+        if dt is not None:
+            tensor = tensor.to(dtype=dt)
+
+        if config is not None:
+            true_tensor = torch.rot90(tensor, **config)
+        else:
+            true_tensor = None
+
+        out_pil_img = F.affine(
+            pil_img, angle=angle, translate=[0, 0], scale=1.0, shear=[0.0, 0.0], interpolation=NEAREST
+        )
+        out_pil_tensor = torch.from_numpy(np.array(out_pil_img).transpose((2, 0, 1))).to(device)
+
+        if scripted_affine:
+            fn = torch.jit.script(F.affine)
+        else:
+            fn = F.affine
+
+        out_tensor = fn(
+            tensor, angle=angle, translate=[0, 0], scale=1.0, shear=[0.0, 0.0], interpolation=NEAREST
+        )
+        if true_tensor is not None:
```

Comment:
let's change that into `if config is not None` and just do

```py
if config is not None:
    assert_equal(torch.rot90(tensor, **config), out_tensor, ...)
```

and we can remove the if/else block about true_tensor above.

No need to pass `msg` either

Reply:
I should remove the `msg` altogether ?

Reply:
done :)